beacon_node/eth2_libp2p/src/discovery/mod.rs

```diff
@@ -727,7 +768,11 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
         };
         // predicate for finding nodes with a matching fork and valid tcp port
         let eth2_fork_predicate = move |enr: &Enr| {
-            enr.eth2() == Ok(enr_fork_id.clone()) && (enr.tcp().is_some() || enr.tcp6().is_some())
+            // `next_fork_epoch` and `next_fork_version` can be different so that
```

Comment:
@AgeManning this might cause compatibility issues with lighthouse nodes that are not altair enabled. 
A non altair enabled node will reject an altair node during discovery because it requires the entire enr fork id to be the same. However, nodes with different `next_fork_epoch` and `next_fork_version`  in their enr fork id should still be accepted according to the spec.

Reply:
Can we just match on either current fork digest or next fork version? 
I think its fine to discover peers who say they are on the next fork digest. After we dial we'll kick them if they are not compatible.

Reply:
Didn't get you. 
I'm saying if node A (old networking) runs a discovery query and finds node B (altair networking), then it will reject node B in the discovery step because node A is running predicate `enr_fork_digest_a == enr_fork_digest_b` which is `false`. So old nodes will not be able to discover altair nodes (other way works fine). Not sure if I'm missing something here.

Reply:
Tested this in a local testnet and it is an issue. The non-altair node rejects the altair nodes. Made a PR to fix the issue here https://github.com/sigp/lighthouse/pull/2433